src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,13 +1,68 @@
 package mate.academy.lib;
 
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.util.HashMap;
+import java.util.Map;
+import mate.academy.service.FileReaderService;
+import mate.academy.service.ProductParser;
+import mate.academy.service.ProductService;
+import mate.academy.service.impl.FileReaderServiceImpl;
+import mate.academy.service.impl.ProductParserImpl;
+import mate.academy.service.impl.ProductServiceImpl;
+
 public class Injector {
     private static final Injector injector = new Injector();
+    private final Map<Class<?>, Class<?>> interfaceToImplementation = new HashMap<>();
+
+    private Injector() {
+        interfaceToImplementation.put(ProductService.class, ProductServiceImpl.class);
+        interfaceToImplementation.put(ProductParser.class, ProductParserImpl.class);
+        interfaceToImplementation.put(FileReaderService.class, FileReaderServiceImpl.class);
+    }
 
     public static Injector getInjector() {
         return injector;
     }
 
     public Object getInstance(Class<?> interfaceClazz) {
-        return null;
+        Class<?> clazz = findImplementation(interfaceClazz);
+        if (!clazz.isAnnotationPresent(Component.class)) {
+            throw new RuntimeException("Class " + clazz.getName()
+                    + " is not annotated with @Component");
+        }
+        Object clazzInstance = createNewInstance(clazz);
+        Field[] declaredFields = clazz.getDeclaredFields();
+        for (Field field : declaredFields) {
+            if (field.isAnnotationPresent(Inject.class)) {
+                Object fieldInstance = getInstance(field.getType());
```

Comment:
This recursive call to `getInstance` could potentially lead to a stack overflow if there are circular dependencies. Consider adding logic to detect and handle circular dependencies.